### DISCUSSION TITLE
How does Modal Interaction group?

### DISCUSSION THREAD

**Discussion by ha-ves** [discord-net/Discord.Net#2558] [Q&A] [answered]
I have this nested slash command module that has a modal response with a `ModalInteraction` in it:
```csharp
[Group(ServerCmd.Server, ServerCmd.ServerDesc)]
internal class GameServerDiscordModule : InteractionModuleBase<SocketInteractionContext> {
...
    [SlashCommand(ServerCmd.Add, ServerCmd.AddDesc)]
    internal async Task ServerAddCmd_Exec(...) {
        await Context.Interaction.RespondWithModalAsync<AddServerModal>(ServerCmd.AddModal + servername);
    }

    [ModalInteraction(ServerCmd.AddModal)]
    internal async Task ServerAddModal_Exec(AddServerModal modal) {
    ...
    }
...
}
```
Is this a correct usage of `ModalInteraction`? Because the `ServerAddModal_Exec` cannot capture the submitted Modal.
I assume because it's inside a group/module it has to have the prefix, right? how do i use it as customid for `RespondWithModalAsync`?

**Comment by cjbonn** [comment]
Add `ignoreGroupNames: true` to the ModalInteraction to ignore group names

**Reply by ha-ves** [reply]
This do capture the command, as with regex and wildcard, but i wanna know how the grouping works, especially if the custom-id is dynamically changed at runtime.

**Comment by Cenngo** [comment, marked as answer]
Interaction Service uses the delimiters in `InteractionServiceConfig.InteractionCustomIdDelimiters` to traverse the nested group structures. So you can assing that prop a value if you want to enable that behaviour. But what @cjbonn said should also work.

**Reply by ha-ves** [reply]
Is there any way to access that Property from a Command Task?

**Reply by ha-ves** [reply]
Ok. This answers my question but needs more explanation on how `ModalInteraction` works, especially when [the docs](https://discordnet.dev/guides/int_framework/intro.html) has yet to mention separators/delimiters.


For any interaction with `custom-id`, it needs to be configured as such:
```csharp
// Chosen delimiters
String[] delimits = new[] { '_' };
```

```csharp
// Creates a InteractionServiceConfig with the delimiters
var interDelimits = new InteractionServiceConfig() { InteractionCustomIdDelimiters = delimits };
// When creating a InteractionService, also pass in the config
var BotInteraction = new InteractionService(discord: BotClient, config: interDelimits);
```

```csharp
// Then when sending responses, use the delimiters
await Context.Interaction.RespondWithModalAsync<AddServerModal>(customid: ServerCmd.Server + delimits[0] + ServerCmd.AddModal);
```
**The rest is covered in [the docs](https://discordnet.dev/guides/int_framework/intro.html)**
